hiku/result.py

```diff
@@ -71,20 +71,25 @@ def ref(self, node, ident):
         return Ref(self.index, node, ident)
 
 
-def _filter_fields(result, node):
-    return {f.name: result[f.name] for f in node.fields}
-
-
-def _denormalize_record(record, result, node):
-    denormalized = {}
-    for f in node.fields:
-        type_ = record.__field_types__[f.name]
-        if isinstance(type_, RecordMeta):
-            denormalized[f.name] = _denormalize_record(type_, result[f.name],
-                                                       f.node)
-        else:
-            denormalized[f.name] = result[f.name]
-    return denormalized
+def _denormalize_type(type_, result, query_node):
+    if isinstance(type_, SequenceMeta):
+        return [_denormalize_type(type_.__item_type__, item, query_node)
+                for item in result]
+    elif isinstance(type_, OptionalMeta):
+        return (_denormalize_type(type_.__type__, result, query_node)
+                if result is not None else None)
+    elif isinstance(type_, RecordMeta):
+        denormalized = {}
+        for f in query_node.fields:
+            field_type = type_.__field_types__[f.name]
+            if isinstance(field_type, TypingMeta):
```

Comment:
This condition here is not necessary, for every `f` from `query_node.fields` we can recursively call `_denormalize_type`, which will handle field type check.

Reply:
Thanks, didn't noticed that on a first glance. Simplified in 0b4c1e9